src/security/PeerConnector.h

```diff
@@ -112,6 +121,13 @@ class PeerConnector: virtual public AsyncJob
 
     /// Called by Downloader after a certificate object downloaded.
     void certDownloadingDone(SBuf &object, int status);
+
+    /// Called while TLS negotiated before the squid-client sends
+    /// the final TLS negotiation messages to the server (eg keys
+    /// and client finished messages) to initiate required callouts
+    /// to external resources (eg downloads missing certificates)
+    /// \return true if required callouts to external resources
+    bool needsValidationCallouts();
```

Comment:
Just to clarify so that I can polish this description: In this context, the word "callouts" is unrelated to the doCallouts() function(ality) in Squid, right? Here, the word "callouts" is used in its more general "queryin an external service" sense, right?

Reply:
Yep. Download certificates and in future I believe we can add external validator call.